nemo_curator/__init__.py

```diff
@@ -29,3 +29,6 @@
 # See https://github.com/NVIDIA/NeMo-Curator/issues/33
 # This also happens when reading and writing to files
 dask.config.set({"dataframe.convert-string": False})
+
+# See setup.py
+__version__ = "0.4.0"
```

Comment:
Thanks for the adding this! One concern I have is needing to manually update this to stay in sync with setup.py. I wonder if there's a way to populate this from our `setup.py` file or perhaps there's a standard canonical approach many python packages follow. I need to read up a bit on this. Curious if perhaps @vyasr has some suggestions here since I'm not too familiar with python versioning/packaging semantics.

Reply:
Yeah there are standard tools here. There are a couple of things I would look at:
1. If your goal is only to make sure that __init__.py and setup.py stay in sync, I would look at [updating this package to specify more of the metadata in pyproject.toml instead of setup.py](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html), at which point you can use [setuptools' dynamic metadata support to read the version from __init__.py](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata).
2. If you want to go one step further and extract the version based on git tags ([it looks like you have some](https://github.com/NVIDIA/NeMo-Curator/tags)) then once you complete step one I would look at https://github.com/pypa/setuptools-scm. It will pull the version from your git tags.

Reply:
Thanks! I ended up only doing step 1 for now. When I tried to implement step 2 (see [this commit](https://github.com/NVIDIA/NeMo-Curator/pull/323/commits/df3550e7b609398840d214a1a06451bc2349fefb)), I was having trouble getting it to grab any tags, both locally and on GitHub CI.

Locally, it would be `nemo_curator-0.0.post181+g5c03c7c.d20241106`, where 181 is the total number of commits we have in the NeMo Curator repository, `5c03c7c` is the latest commit hash, and `20241106` is the date. Since it was `nemo_curator-0.0`, it was not capturing the 0.5.0 tag.

I saw similar behavior on GitHub CI, see an [example run](https://github.com/NVIDIA/NeMo-Curator/actions/runs/11712284148/job/32622597310) here. If you navigate to "Install NeMo-Curator and pytest", you will see `Collecting awscli>=1.22.55 (from nemo_curator==0.0.post195+g74dbc65)`, where this time 195 is the total number of commits in the NeMo Curator repository plus the number of commits in this PR at the time.

For now, I think it should be okay to just have the version in a single location, in our `pyproject.toml` file.

Reply:
Yes when I did `git fetch --tags` they were there. Very strange behavior IMO, I tried various things but couldn't get it to work at all for me.

Reply:
Strange. Anyway, if you're happy with the current state we're good. If you want help setting up setuptools-scm or similar later I can take a look another time.